l3utils.lua

```diff
@@ -0,0 +1,79 @@
+--[[
+
+File l3utils.lua Copyright (C) 2014-2017 The LaTeX3 Project
+
+It may be distributed and/or modified under the conditions of the
+LaTeX Project Public License (LPPL), either version 1.3c of this
+license or (at your option) any later version.  The latest version
+of this license is in the file
+
+   http://www.latex-project.org/lppl.txt
+
+This file is part of the "l3build bundle" (The Work in LPPL)
+and all files in that bundle must be distributed together.
+
+-----------------------------------------------------------------------
+
+The development version of the bundle can be found at
+
+   https://github.com/latex3/latex3
+
+for those people who are interested.
+
+--]]
+
+-- the LaTeX3 utils namespace
+local l3utils = {}
+
+--- Checks if the script is running on a Windows machine.
+-- This function checks if Windows is the underlying operating system
+-- by inspecting the path separator. The occurrence of '\' indicates a
+-- Windows machine.
+-- @return a logic value indicating whether the script is running on a
+-- Windows machine.
+function l3utils.windows()
+  return package.config:sub(1,1) == '\\'
```

Comment:
In `texlua` (at least) we can test `os.type == "windows"` rather than needing to look at `package.config`.